### PR TITLE
drop missing keys from KeyValueRender

### PR DESCRIPTION
**Pull request by sibson**
Rather than render as None, simply drop any missing keys

### CODE REVIEW

**Comment by hynek** [comment]
Hmmmm I feel like this should be a separate processor, such that other renderers benefit from it too.  Would you agree?

**Comment by sibson** [comment]
Are you suggesting something like 

```
def drop_none_processor(event_dict):
    return dict(k: v for k, v in event_dict if v is not None)
```
If so the downside is that it would prevent logging None when its a valid value.  Otherwise, I don't understand your suggestion.

My particular problem is that KeyValueRenderer is injecting spurious new keys into my log messages when I set a key_order.  For example, web requests will have method,path keys which are never present in celery tasks but because Django shares the config I end up with method=None, path=None in my task logs.  I could use different config for each but that seems less DRY..

**Comment by hynek** [comment]
Ah I see; sorry.  I thought you wanted to drop Nones altogether.

File: src/structlog/processors.py [hunk 1]
```diff
@@ -31,27 +31,31 @@ class KeyValueRenderer(object):
...
-    def __init__(self, sort_keys=False, key_order=None):
+    def __init__(self, sort_keys=False, key_order=None, drop_missing=False):
         # Use an optimized version for each case.
         if key_order and sort_keys:
             def ordered_items(event_dict):
                 items = []
                 for key in key_order:
                     value = event_dict.pop(key, None)
-                    items.append((key, value))
+                    if value or not drop_missing:
```
Comment (hynek) [inline src/structlog/processors.py]: This has to be `value is not None` or else `0` and `""` are dropped too.

File: src/structlog/processors.py [hunk 2]
```diff
@@ -31,27 +31,31 @@ class KeyValueRenderer(object):
...
+                        items.append((key, value))
                 items += sorted(event_dict.items())
                 return items
         elif key_order:
             def ordered_items(event_dict):
                 items = []
                 for key in key_order:
                     value = event_dict.pop(key, None)
-                    items.append((key, value))
+                    if value or not drop_missing:
```
Comment (hynek) [inline src/structlog/processors.py]: same

**Comment by sibson** [comment]
@hynek updated to fix dropping "" and 0, thanks

**Comment by hynek** [comment]
thank you!

**Pull request closed without merging**